### ISSUE TITLE
New Party Badges!

### ISSUE THREAD

**Issue by lemoness** [HabitRPG/habitica#6766]
We've got two new achievement badges to implement! Hooray :)

The first is called "Party Up" and is awarded to anyone who has formed a party with at least two total people.

![partybadge1b](https://cloud.githubusercontent.com/assets/3494081/13303180/9ba1acec-db02-11e5-8a8f-f3c27b0bbcec.png)

Description: Joined a Party with another person! Have fun battling monsters and supporting each other.

The second is called "Party On" and is awarded to anyone who has formed a party with at least four total people.

![partybadge2](https://cloud.githubusercontent.com/assets/3494081/13304446/85cfd77a-db09-11e5-98ef-caed50bf1ac4.png)

Description: Was in a Party with at least four people. Enjoy your increased accountability as you unite with your friends to vanquish your foes!

A note that's open to debate -- we should decide whether or not leaving the party causes you to lose the badge. My instinct is that no, once you've earned it you should get to keep it, but I'd like to hear discussion from others.

**Comment by JerseyEmerald** [comment]
I agree , you should be able to keep it but there should be a guidelines. For example, you must keep the party opened for at least 6 months in order to prevent abandonment issues. What about those who already started their own party, do they automatically get this badge...

**Comment by ghost** [comment]
An oddity I see with the debate about keeping it if you leave the party is WikiFAQ_bot. People could get the achievement from it with no intent to be social. (Though, people could team up for that anyway, if they reeeeally wanted an imaginary internet trophy. ;) )

**Comment by JerseyEmerald** [comment]
I see the problem,  maybe it can be implemented in a special quest

**Comment by lemoness** [comment]
People who already have a party with at least two/four users would
definitely need to get the badge.

I'm not too bothered by the WikiFAQbot workaround - after all, inviting it
to a party grants the Basi-List scroll, which is another social incentive
feature. I imagine that would affect a relatively small number of users.

**Comment by JerseyEmerald** [comment]
True, as a newbie myself, I never knew about the WikiFAQbot and it would be a great incentive to work towards. It would also encourage  the Players to be more social in order to obtain it. I think it's a great idea.

**Comment by querry43** [comment]
I don't mind rewarding people for being clever.  Whether they create a party with people or with the bot, they are still engaging in activities on the site and that's the point, right?

**Comment by citrusella** [comment]
I second querry43's sentiment. I got the "gave [certain type of card]" achievements (and [turned into x] during the festivals) by doing it to myself. If a multi-member party isn't compatible with a certain user's playstyle at a given time (like me, I'd happily be in a party with others... just not yet), I don't think they should be barred from one thing intended for multi-member parties and not all the others.

And they'd still have to work with actual people to get the 4-member one. ;P

**Comment by khipkin** [comment]
@lemoness I would like to start working on this!

Also, since I'm commenting anyway, I think that achievements should never be revoked once they've been earned.  It would set a weird precedent.

**Comment by Alys** [comment]
@khipkin Thank you! Comment here if have any questions or would like advice.

For simplicity and to match most of the comments here, lets make it work like this:
- [ ] You gain the achievement as soon as the party you are in gains a new member to bring it up to two/four people, with no need to wait (waiting for a certain time to ensure that the party stays together would require some tricky code and would disadvantage people who found themselves in a party with incompatible people).
- [ ] All players in the party gain the achievement when the party reaches the correct size, including the newest player.
- [ ] Once you have the achievement, you cannot lose it.
- [ ] There's no special treatment for WikiFAQ_bot - inviting it does the same thing as inviting a normal player.

Also, once this is ready to go live, we will need to award the badge to all players who are already in a party with at least two/four users. I will do that because it will need direct access to the database. @khipkin You don't need to do anything for this aspect (although if you are familiar with mongo update commands and want to write a script for this, you'd be welcome to, but don't feel that you have to!).

After this is live, we'll probably get questions from players who used to be in a party of two/four but are not now, wanting to know if they can be given the achievement. I guess we should decide in advance what our response should be. There's no way that we can automatically give the achievement for parties that no longer exist, or players who are no longer in parties, so we'd have to either not do it at all, or award it manually to anyone who asks, trusting them to not be misleading us about their previous party. I think I'm in favour of saying that no, the achievement isn't backdated for anyone - that's simpler and easier. However if we decide that we should give it to anyone who asks, it won't be that hard for a database admin to do (me and a couple of others) - less than a minute per user.

**Comment by JerseyEmerald** [comment]
InI totally agree,  nice job and it's exciting to have a new badge to look forward too!!

**Comment by lemoness** [comment]
I am also in favor of no backdating. Much simpler on our end, and besides,
they can always form a new party to get it!

**Comment by SabreCat** [comment]
Hopping in to subscribe to notifications on this topic. @khipkin, let me know if there's anything you need a hand with!

**Comment by khipkin** [comment]
@Alys @SabreCat How do I include the little achievement pictures in the achievement modals?  I think that's what the [achievementAvatar mixin](https://github.com/HabitRPG/habitrpg/blob/34057c375bf66235d610e08ea2b5007dec4243e1/website/views/shared/modals/achievements.jade#L14) is for... if so, what strings should I use for the "Party Up" and "Party On" badges?

Also, the descriptions for the achievements are inconsistent.  I think they should either both start with "Joined a Party with..." or "Was in a Party with..."

**Comment by lemoness** [comment]
They can both start with "Joined a party with..."

**Comment by khipkin** [comment]
@lemoness @Alys Should your user profile show these achievements as grayed-out before they are earned?  If so, in what position in the list?

**Comment by khipkin** [comment]
I added them right after triad bingo ([here](https://github.com/HabitRPG/habitrpg/blob/34057c375bf66235d610e08ea2b5007dec4243e1/website/views/shared/profiles/achievements.jade#L117)) so that they're at the end of the list, but still before the quests completed and challenges won.

Obviously, I'll move them if that position isn't right. :smile:

**Comment by lemoness** [comment]
Yes, they should show as greyed-out when you don't have them yet! Let's
move them above Beast Master, since pets don't unlock until level 3 but you
can form a party immediately.

**Comment by khipkin** [comment]
That makes sense! Done.

**Comment by Alys** [comment]
For the record, the migration that I described above is no longer desirable because we've told people that they should claim their badges by going to the party page, and if the migration was run, they wouldn't get the notification of the badges when they visited the page.
